### PR TITLE
Issue/test editable installs

### PR DESCRIPTION
**Pull request by edvgui**
# Description

Test editable installs.  cf. https://github.com/inmanta/pytest-inmanta-lsm/pull/312#discussion_r1182216046

# Self Check:

Strike through any lines that are not applicable (`~~line~~`) then check the box

- [x] Attached issue to pull request (no ticket)
- [x] Changelog entry (no change in pytest-inmanta itself)
- [x] Code is clear and sufficiently documented
- [x] Sufficient test cases (reproduces the bug/tests the requested feature)
- [x] Correct, in line with design
- [x] End user documentation is included or an issue is created for end-user documentation (add ref to issue here: )

### CODE REVIEW

File: tests/test_basic_example_v2.py
Comment (sanderr) [inline tests/test_basic_example_v2.py:1]: Actually, with the parameter you added to the context manager we don't even need to make changes to this file, do we? We have consistency between pytest-inmanta and pytest-inmanta lsm in the context manager, we just use it slightly differently here so we can use the session-scoped fixture?

File: tests/test_basic_example_v2.py [hunk 1]
```diff
@@ -34,13 +34,26 @@
...
+        "Tests using the testmodulev2 should have a marker specifying whether the module "
+        "should be installed in editable mode or not."
+    )
+
+
+@pytest.fixture(scope="function")
+def testmodulev2_venv(
+    testdir: pytest.Testdir,
+    pytestconfig: pytest.Config,
+    testmodulev2_editable_install: bool,
```
Comment (sanderr) [inline tests/test_basic_example_v2.py:49]: Wow, does this get its value from the marker of the test that requires the fixture?

File: tests/test_basic_example_v2.py
Comment (edvgui) [reply tests/test_basic_example_v2.py:1]: We could, it just mean that we don't test modules installed in editable mode then

Comment (sanderr) [reply tests/test_basic_example_v2.py:1]: Good point. For pytest-inmanta I don't think it should make a difference but it's good to cover both cases. If it doesn't have too big an impact on test time I'm all for it.

**Review by sanderr — APPROVED** [review]

**Pull request merged**